.github/workflows/test.yml

```diff
@@ -0,0 +1,81 @@
+name: Tests / Code Coverage
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, labeled]
+  merge_group:
+    types: [checks_requested]
+
+permissions:
+  contents: read
+
+concurrency:
+  group: ci-${{ github.ref }}-tests
+  cancel-in-progress: true
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        arch: [amd64, arm64]
+        targetos: [darwin, linux]
+    name: sedad ${{ matrix.targetos }}-${{ matrix.arch }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: Cache binaries
+        id: cache-binaries
+        uses: actions/cache@v3
+        with:
+          path: ./cmd/sedad/sedad
+          key: sedad-${{ matrix.targetos }}-${{ matrix.arch }}
+      - uses: technote-space/get-diff-action@v6.1.2
+        with:
+          PATTERNS: |
+            **/**.go
+            go.mod
+            go.sum
+      - name: Setup go
+        if: steps.cache-binaries.outputs.cache-hit != 'true' && env.GIT_DIFF
+        uses: actions/setup-go@v4
+        with:
+          go-version: "1.21"
```

Comment:
1.19? perhaps doesn't matter

Reply:
good point .50 has go updated but I will change it then thanks!